src/main/java/inf112/isolasjonsteamet/roborally/board/BoardClientImpl.java

```diff
@@ -0,0 +1,116 @@
+
+/**
+ * Class for our boards that hold all the client related code (libgdbx).
+ */
+public class BoardClientImpl extends BoardImpl {
+
+	public TiledMap map;
+	public TiledMapTileLayer boardLayer;
+	public TiledMapTileLayer playerLayer;
+	public TiledMapTileLayer holeLayer;
+	public TiledMapTileLayer flagLayer;
+
+	public TiledMapTileLayer.Cell playerWonCell;
+	public TiledMapTileLayer.Cell playerDiedCell;
+	public TiledMapTileLayer.Cell playerCell;
+	public TiledMapTileLayer.Cell transparentCell;
+
+	/**
+	 * Create a new instance of BoardClientImpl.
+	 */
+	public BoardClientImpl(List<Player> players, String boardFilename) {
+		super(players, new ArrayList<>());
+		map = new TmxMapLoader().load(boardFilename);
+		loadCells();
+		tiles = tilesFromMap(boardFilename);
+		height = playerLayer.getHeight();
+		width = playerLayer.getWidth();
+	}
+
+	/**
+	 * Load cells needed to build the board.
+	 */
+	private void loadCells() {
+		boardLayer = (TiledMapTileLayer) map.getLayers().get("Board");
+		playerLayer = (TiledMapTileLayer) map.getLayers().get("Player");
+		holeLayer = (TiledMapTileLayer) map.getLayers().get("Hole");
+		flagLayer = (TiledMapTileLayer) map.getLayers().get("Flag");
+
+		Texture playerTx = new Texture("player.png");
+		final TextureRegion[][] tReg = TextureRegion.split(playerTx, 300, 300);
+
+		Texture defTx = new Texture("tiles.png");
+		final TextureRegion[][] tReg2 = TextureRegion.split(defTx, 300, 300);
+
+		//Static tiles for playing, dead and winning player cells
+		var transparentTile = new StaticTiledMapTile(tReg2[15][4]);
+		transparentCell = new TiledMapTileLayer.Cell().setTile(transparentTile);
+
+		var playerTile = new StaticTiledMapTile(tReg[0][0]);
+		playerCell = new TiledMapTileLayer.Cell().setTile(playerTile);
+
+		var playerWonTile = new StaticTiledMapTile(tReg[0][2]);
+		playerWonCell = new TiledMapTileLayer.Cell().setTile(playerWonTile);
+
+		var playerDiedTile = new StaticTiledMapTile(tReg[0][1]);
+		playerDiedCell = new TiledMapTileLayer.Cell().setTile(playerDiedTile);
+	}
+
+	/**
+	 * Get tiles from the map.
+	 */
+	private ImmutableList<List<List<TileType>>> tilesFromMap(String boardFilename) {
+		int width = playerLayer.getWidth();
+		int height = playerLayer.getHeight();
+
+		List<List<List<TileType>>> accX = new ArrayList<>(width);
+		for (int x = 0; x < width; x++) {
+			List<List<TileType>> accY = new ArrayList<>(height);
+			for (int y = 0; y < height; y++) {
+				List<TileType> acc = new ArrayList<>();
+
+				if (boardLayer.getCell(x, y) != null) {
+					acc.add(Tiles.GROUND);
+				}
+
+				if (holeLayer.getCell(x, y) != null) {
+					acc.add(Tiles.HOLE);
+				}
+
+				if (flagLayer.getCell(x, y) != null) {
+					acc.add(Tiles.FLAG);
+				}
+
+				accY.add(ImmutableList.copyOf(acc));
+			}
+			accX.add(ImmutableList.copyOf(accY));
+		}
+
+		return ImmutableList.copyOf(accX);
+	}
+
+	/**
+	 * Update the playerview.
+	 */
+	public void updatePlayerView() {
+		for (Player player : players) {
```

Comment:
This method won't work as well if we move the player start position in the map file. Can we remove the player from the map file?